File: manager_web/src/api/platformConfig.ts
import { apiClient } from './config';

// 类型定义
export interface PlatformConfig {
  _id: string;
  platform_id: string;
  platform_name: string;
  app_key: string;
  app_secret: string;
  pid: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface PlatformConfigCreate {
  platform_id: string;
  platform_name: string;
  app_key?: string;
  app_secret?: string;
  pid?: string;
  is_active?: boolean;
}

export interface PlatformConfigUpdate {
  app_key?: string;
  app_secret?: string;
  pid?: string;
  is_active?: boolean;
}

export interface SupportedPlatform {
  platform_id: string;
  platform_name: string;
  icon: string;
  description: string;
  doc_url: string;
}

// API 方法
export const platformConfigApi = {
  // 获取平台配置（不传platform_id返回所有，传则返回单个）
  getConfigs: async (platformId?: string): Promise<PlatformConfig[]> => {
    const params = platformId ? { platform_id: platformId } : {};
    const response = await apiClient.get<PlatformConfig[]>('/platform-configs', { params });
    return response.data;
  },

  // 获取支持的平台列表
  getSupportedPlatforms: async (): Promise<SupportedPlatform[]> => {
    const response = await apiClient.get<SupportedPlatform[]>('/platform-configs/supported-platforms');
    return response.data;
  },

  // 获取已启用的平台
  getActiveConfigs: async (): Promise<PlatformConfig[]> => {
    const response = await apiClient.get<PlatformConfig[]>('/platform-configs/active');
    return response.data;
  },

  // 获取已配置的平台
  getConfiguredConfigs: async (): Promise<PlatformConfig[]> => {
    const response = await apiClient.get<PlatformConfig[]>('/platform-configs/configured');
    return response.data;
  },

  // 创建平台配置
  createConfig: async (data: PlatformConfigCreate): Promise<PlatformConfig> => {
    const response = await apiClient.post<PlatformConfig>('/platform-configs', data);
    return response.data;
  },

  // 更新平台配置
  updateConfig: async (platformId: string, data: PlatformConfigUpdate): Promise<PlatformConfig> => {
    const response = await apiClient.put<PlatformConfig>(`/platform-configs?platform_id=${platformId}`, data);
    return response.data;
  },

  // 切换平台启用状态
  toggleActive: async (platformId: string, isActive: boolean): Promise<PlatformConfig> => {
    const response = await apiClient.patch<PlatformConfig>(`/platform-configs/active?platform_id=${platformId}&is_active=${isActive}`);
    return response.data;
  },

  // 删除/清空平台配置
  deleteConfig: async (platformId: string): Promise<{ message: string; success: boolean }> => {
    const response = await apiClient.delete(`/platform-configs?platform_id=${platformId}`);
    return response.data;
  },
};

File: server/app/models/platform_config.py
"""
电商平台配置模型
"""
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class PlatformConfigBase(BaseModel):
    """平台配置基础模型"""
    platform_id: str = Field(..., description="平台标识: alimama/jd/pdd")
    platform_name: str = Field(..., description="平台名称")
    app_key: str = Field(default="", description="AppKey")
    app_secret: str = Field(default="", description="AppSecret (加密存储)")
    pid: str = Field(default="", description="推广位PID (用于佣金追踪)")
    is_active: bool = Field(default=False, description="是否启用")


class PlatformConfigCreate(PlatformConfigBase):
    """创建平台配置请求模型"""
    pass


class PlatformConfigUpdate(BaseModel):
    """更新平台配置请求模型"""
    app_key: Optional[str] = None
    app_secret: Optional[str] = None
    pid: Optional[str] = None
    is_active: Optional[bool] = None


class PlatformConfigResponse(PlatformConfigBase):
    """平台配置响应模型"""
    id: str = Field(..., alias="_id")
    created_at: datetime
    updated_at: datetime

    class Config:
        populate_by_name = True


class PlatformConfig(PlatformConfigBase):
    """平台配置完整模型"""
    id: str = Field(..., alias="_id")
    created_at: datetime
    updated_at: datetime

    class Config:
        populate_by_name = True


# 支持的平台列表
SUPPORTED_PLATFORMS = [
    {
        "platform_id": "alimama",
        "platform_name": "阿里妈妈",
        "icon": "🅰️",
        "description": "淘宝客API，提供商品搜索、佣金查询、订单跟踪等功能",
        "doc_url": "https://pub.alimama.com/",
    },
    {
        "platform_id": "jd",
        "platform_name": "京东联盟",
        "icon": "🅹",
        "description": "京东联盟API，提供商品推广、佣金结算、订单查询等功能",
        "doc_url": "https://union.jd.com/",
    },
    {
        "platform_id": "pdd",
        "platform_name": "多多客",
        "icon": "🅿️",
        "description": "拼多多多多客API，提供商品推广、佣金查询、订单跟踪等功能",
        "doc_url": "https://www.pinduoduo.com/",
    },
]
